ocaml/xenops/domain.ml

```diff
@@ -211,15 +213,26 @@ let make ~xc ~xs vm_info uuid =
 			(* create cpu and memory directory with read only perms *)
 			List.iter (fun dir ->
 				let ent = sprintf "%s/%s" dom_path dir in
-				t.Xst.mkdir ent;
-				t.Xst.setperms ent roperm
+				mkdirperms t ent roperm
 			) [ "cpu"; "memory" ];
+
+
+			let mksubdirs base dirs perms =
+				List.iter (fun dir ->
```

Comment:
I think that when backporting from trunk to a maintenance branch, we should copy without changes where applicable.
Therefore given that this bit of code is cherry-picked from trunk onto this maintenance branch, I think we should keep it as it is.